### PR TITLE
Update smd to v4

### PR DESCRIPTION
**Pull request by apelisse**
Update SMD to use an actual tag. I've also created the branch and everything so that it's ready if we have to make a cherry-pick for that in the future.

Note that it's using 4.0.1 because I initially pushed 4.0.0 to the wrong commit, and then it looks like something is unhappy if you force a tag :-)

Beside that, it should be a no-op.

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -498,6 +498,7 @@ replace (
 	sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.9
 	sigs.k8s.io/kustomize => sigs.k8s.io/kustomize v2.0.3+incompatible
 	sigs.k8s.io/structured-merge-diff/v3 => sigs.k8s.io/structured-merge-diff/v3 v3.0.1-0.20200706213357-43c19bbb7fba
+	sigs.k8s.io/structured-merge-diff/v4 => sigs.k8s.io/structured-merge-diff/v4 v4.0.1
```
Comment (apelisse) [inline go.mod:501]: There must be something wrong if I still have v3 above.

Comment (apelisse) [reply go.mod:501]: Yeah, looks like I missed a few files

**Comment by liggitt** [comment]
Things targeting 1.19 should still go to master, it hasn't reopened for 1.20 yet

**Comment by liggitt** [comment]
Closing in favor of https://github.com/kubernetes/kubernetes/pull/93867

**Pull request closed without merging**